docs/java-your-first-app.md

```diff
@@ -0,0 +1,163 @@
+
+In case you are still wondering what [Temporal](/docs/overview) is, that is OK, there is a lot that can be said about it. In this tutorial we will focus on two of its main user facing components which are the Temporal server and its Java SDK.
+
+![Temporal server and SDK diagram](/img/docs/temporal-server-and-sdk-icons.png)
+
+Together they are a solution to the complexities that arise from modern application development needs. You can think of Temporal as a sort of "cure all" for the pains developers experience from having to implement retry logic, timeout logic, databases to track application states, rollbacks due to process failures, and more.
+
+This tutorial is aimed at developers who are new to Temporal and have some basic knowledge of Java. By following this tutorial you will learn how to run a Temporal Workflow application using the [Temporal Java SDK](https://github.com/temporalio/java-sdk) and you will also witness the fundamentally resilient nature of a Workflow, getting just a taste of the value that Temporal can provide.
+
+## Project setup
+
+![Tool box](/img/docs/repair-tools.png)
+
+First, make sure you have the [tutorial prerequisites](/docs/java-sdk-tutorial-prerequisites) setup.
+
+Next, download the ['java-project-template'](https://github.com/temporalio/java-project-template) by downloading it as a zip or by [creating a new repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-from-a-template#creating-a-repository-from-a-template) in your own Github account and then cloning the repo to the location of your choice.
+
+The last setup step is to build the project. If you open it with [IntelliJ](https://www.jetbrains.com/idea/) the project will build automatically. If you prefer to run programs from your terminal, make sure you have [Gradle](https://gradle.org/install/) installed and run the Gradle build command from the root of the project:
+
+```
+./gradlew build
+```
+
+Once your project has built, you are ready to go.
+
+## Application overview
+
+![Tool box](/img/docs/workflow.png)
+
+The Java project template mimics a "money transfer" application in order to give you the minimum elements needed to both get started building your own application as well as understand some of the value that Temporal gives you right out of the box. The project includes a predefined [Workflow](/docs/workflows) that orchestrates the execution of an account object's methods `withdraw()` and `deposit()` to mock the transfer of money from one account to another. Temporal calls such methods [Activities](/docs/activities).
+
+The first thing you will do is to call on the Temporal server to start the transfer. Temporal will then start tracking the progress of the Workflow execution. You will then run a Worker whose sole job is to execute the functions of the Workflow and communicate the results back to the Temporal server.
```

Comment:
I think this could be better explained. I like the high level ordering of the paragraphs but once I get to the worker part its a bit confusing.

Reply:
OK, gave it a refresh.